src/services/generalServices.js

```diff
@@ -220,137 +337,17 @@ const exportToExcel = async (initialState, updatedState, prevSeasonYear, currSea
-                {key: 'stateInfoNames', width: 42.5, style: { font: { name: 'Calibri' }}},
-                {key: 'initialStateInfo', width: 10, style: { font: { name: 'Calibri' }}},
-                {key: 'to_state', width: 10, style: { font: { name: 'Calibri' }}},
-                {key: 'updatedStateInfo', width: 10, style: { font: { name: 'Calibri' }, alignment: {vertical: 'bottom', horizontal: 'right'}}},
-                {key: 'empty_1', width: 10, style: { font: { name: 'Calibri' }}},
-                {key: 'empty_2', width: 10, style: { font: { name: 'Calibri' }}},
-                {key: 'regionName', width: 25, style: { font: { name: 'Calibri' }}},
-                {key: 'regionIncome', width: 25, style: { font: { name: 'Calibri' }, alignment: {vertical: 'bottom', horizontal: 'right'}}},
-                {key: 'foodProduced', width: 25, style: { font: { name: 'Calibri' }}},
-                {key: 'populationUsed', width: 25, style: { font: { name: 'Calibri' }}},
-                {key: 'currPopulation', width: 25, style: { font: { name: 'Calibri' }, alignment: {vertical: 'bottom', horizontal: 'right'}}},
-                {key: 'maxPopulation', width: 25, style: { font: { name: 'Calibri' }}},
-            ]
-            let facilityCount = await facility.getFacilityCountByStateId(updatedState[i].stateID)
-            let adminCost = await stateServices.getAdminCostByStateId(updatedState[i].stateID);
-            let updatedExpectedIncome = parseFloat(parseFloat(updatedState[i].TotalIncome).toFixed(2) - parseFloat(updatedState[i].expenses).toFixed(2) - parseFloat(adminCost).toFixed(2)).toFixed(2);
-
-            initialStateInfo = [
-                initialState[i].treasuryAmt,
-                'N/A',
-                'N/A',
-                'N/A',
-                'N/A',
-                'N/A',
-                'N/A',
-                'N/A',
-                'N/A',
-                initialState[i].TotalPopulation,
-                'N/A',
-                'N/A',
-                'N/A',
-            ]
-            updatedStateInfo = [
-                updatedState[i].treasuryAmt,
-                updatedState[i].TotalIncome,
-                updatedState[i].expenses,
-                adminCost,
-                updatedState[i].adminRegionModifier * 100 + '%',
-                adminCost + updatedState[i].expenses,
-                updatedState[i].TotalFoodProduced,
-                updatedState[i].TotalFoodConsumed,
-                updatedState[i].TotalFoodAvailable,
-                updatedState[i].TotalPopulation,
-                updatedState[i].AvgDevLevel,
-                facilityCount,
-                updatedExpectedIncome,
-            ]
+            sheet.columns = await initializeExcelColumns();
+
+            let stateInfos = await populateStateInfo(initialState, updatedState, i)
+            initialStateInfo = stateInfos[0];
+            updatedStateInfo = stateInfos[1];
+
             //Make title for each sheet with stylings

```

Comment:
Instead of commenting here, how about grouping all these into one function, something like `titleEachSheet`?

Reply:
idt there's a need for that. I'd need to pass more things by value (the sheet, and the state object again). This is already clear enough imo